### PR TITLE
capInitialTargetBatchSizeByMaxBatchSize

### PR DESCRIPTION
**Pull request by xinlian12**
Issue:
`new CosmosBulkExecutionOptions().setMaxMicroBatchSize(maxBatchSize);` - when setting maxBatchSize < initialBatchSize, the expected behavior would be all the mini batch size <= maxBatchSize, but currently the first mini batch still uses the initialBatchSize.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #### Breaking Changes
 
 #### Bugs Fixed
+* Fixed an issue where initial `targetBatchSize` is larger than `maxBatchSize` configured in `CosmosBulkExecutionOptions` - See[39500](https://github.com/Azure/azure-sdk-for-java/pull/39500)
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:10]: NIT - i would  classify this as "Other" - you can argue either way what the expected way should be. And honoring the config is not really a bug?

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: updated

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - thanks!

**Review by kushagraThapar — APPROVED** [review]
LGTM, thanks @xinlian12

**Pull request merged**